subsys/net/ip/icmpv4.c

```diff
@@ -26,11 +26,93 @@
 
 static sys_slist_t handlers;
 
+struct net_icmp_hdr *net_icmpv4_set_hdr(struct net_pkt *pkt,
+					struct net_icmp_hdr *hdr)
+{
+	struct net_buf *frag = pkt->frags;
+	struct net_icmp_hdr *icmp_hdr;
+	u16_t pos;
+
+	icmp_hdr = net_pkt_icmp_data(pkt);
+	if (net_check_icmp_header(pkt, icmp_hdr)) {
```

Comment:
Suppose confusingly named net_check_icmp_header() got renamed, or I just figured out what it actually does. But how to explain the fact that if ICMP header gets split between fragments, the we diligently copy values from input ICMP header `hdr` into packet (below), while if it fits, we don't do anything?